src/components/sidebar-navigation/SidebarNavigationHeader.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import styled from '@emotion/styled';
+import { forwardRef, ReactNode, useRef } from 'react';
+import { truncate } from '~common/helpers/styles';
+import { useIsOverflow } from '~common/helpers/useIsOverflow';
+import { TextNode, TextNodeOptional } from '~types/utils';
+import { cssVar } from '~utils/design-tokens';
+import { IconExpandAll } from '../icons';
+import { Tooltip, TooltipSide } from '../tooltip';
+import { Text } from '../typography';
+
+export interface SidebarNavigationHeaderProps {
+  /**
+   * Image to display to the left of the header.
+   */
+  avatar?: ReactNode;
+  /**
+   * Display a Icon on the right, indicating it triggers a dropdown.
+   * To be set to `true` when used as a dropdown trigger.
+   */
+  isInteractive?: boolean;
+  /**
+   * Text to display under the main text of the header.
+   * Typically defines the type of entity being shown (organization, enterprise, project, ...)
+   */
+  qualifier?: TextNodeOptional;
+  /**
+   * The main text to show in the header
+   */
+  name: TextNode;
+}
+
+export const SidebarNavigationHeader = forwardRef<HTMLButtonElement, SidebarNavigationHeaderProps>(
```

Comment:
I'm wondering about the name, if it should be `SidebarHeader` instead?
For example for the accordion I used `SidebarAccordionItem`, since it's not doing navigation.